server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -2523,6 +2523,12 @@ public void run() {
         Assignment assignment = new Assignment(extent, getTabletSession());
         TabletStateStore.setLocation(getContext(), assignment);
```

Comment:
I am wondering if this might be a better place to set the last location... if you look at what it calls, you will get to ```MetaDataStateStore.setLocations()``` and [here](https://github.com/apache/accumulo/blob/e22ea93786281324702e36b1dc0c151a6be1fa7c/server/base/src/main/java/org/apache/accumulo/server/master/state/MetaDataStateStore.java#L68) you might just add this one line:
```assignment.server.putLastLocation(m);```

This will include it in the mutation that sets the location.  It will most likely set the last location the same as the location but I don't think that we care at this point.  It would help in your case where a Tablet has been unassigned.

Reply:
Okay, I can try that when I get in Monday. Thanks for the suggestion.

Reply:
This method seems to work pretty well. There are some instances where it is brought back up in a slightly unbalanced state but it is very minimal. Taking that and performance into consideration, this method is probably the most ideal option all around.